packages/paste-core/components/progress-bar/stories/index.stories.tsx

```diff
@@ -0,0 +1,137 @@
+import * as React from 'react';
+import {Label} from '@twilio-paste/label';
+import {HelpText} from '@twilio-paste/help-text';
+import {useUID} from '@twilio-paste/uid-library';
+import {Form, FormControl} from '@twilio-paste/form';
+import {Button} from '@twilio-paste/button';
+import {Box} from '@twilio-paste/box';
+import {CustomizationProvider} from '@twilio-paste/customization';
+
+import {ProgressBar} from '../src';
+
+// eslint-disable-next-line import/no-default-export
+export default {
+  title: 'Components/ProgressBar',
+};
+
+const NumberFormatter = new Intl.NumberFormat('en-US');
+
+export const Default = (): React.ReactNode => {
+  const progressBarId = useUID();
+  const helpTextId = useUID();
+  const [value, setValue] = React.useState<number>(0);
+  const [rerun, setRerun] = React.useState<number>(1);
+
+  // Randomly updates the value of the progress bar to simulate a real progress bar
+  React.useEffect(() => {
+    let interval;
+    if (rerun) {
+      interval = setInterval(() => {
+        setValue((previousValue) => {
+          if (previousValue >= 100) {
+            setRerun(0);
+            return 100;
+          }
+          return previousValue + Math.random() * 12;
+        });
+      }, 600);
+    }
+    return () => clearInterval(interval);
+  }, [rerun]);
+
+  return (
+    <Box maxWidth="600px">
+      <Form>
+        <FormControl>
+          <Label htmlFor={progressBarId}>Downloading more ram</Label>
+          <ProgressBar id={progressBarId} aria-describedby={helpTextId} value={value} />
+          <HelpText id={helpTextId}>
+            Increasing your ram helps your computer run faster. {NumberFormatter.format(value)}% complete.
+          </HelpText>
+        </FormControl>
+        <Button
+          variant="primary"
+          onClick={() => {
+            setRerun(1);
+            setValue(0);
+          }}
+          disabled={rerun === 1}
+        >
+          Restart Progress
+        </Button>
+      </Form>
+    </Box>
+  );
+};
+
+export const Indeterminate = (): React.ReactNode => {
+  const progressBarId = useUID();
+  const helpTextId = useUID();
+
+  return (
+    <Box maxWidth="600px">
+      <Form>
+        <FormControl>
+          <Label htmlFor={progressBarId}>Downloading more ram</Label>
+          <ProgressBar id={progressBarId} aria-describedby={helpTextId} isIndeterminate />
+          <HelpText id={helpTextId}>Increasing your ram helps your computer run faster.</HelpText>
+        </FormControl>
+      </Form>
+    </Box>
+  );
+};
+
+export const LabelingApproaches = (): React.ReactNode => {
+  const labelId = useUID();
+
+  return (
+    <Box maxWidth="600px">
+      <Form>
+        <FormControl>
+          <Box id={labelId}>Using aria-labelledby demo</Box>
```

Comment:
suggestion: these should all be "Labels" in the examples, showing everyone the best practices. But obviously the `ProgressBarLabel` that you will create.